### ISSUE TITLE
WARNING: possible circular locking dependency detected

### ISSUE THREAD

**Issue by marcin-github** [koverstreet/bcachefs#112]
Hi,

```
[ 1334.986239] ======================================================
[ 1334.986490] WARNING: possible circular locking dependency detected
[ 1334.986788] 5.6.0-134515-ge1f6739c4a9f #21 Tainted: G           OE   T
[ 1334.987109] ------------------------------------------------------
[ 1334.987450] kswapd0/109 is trying to acquire lock:
[ 1334.988014] ffff88820660b570 (&wp->lock){+.+.}, at: bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1334.988442]
               but task is already holding lock:
[ 1334.988837] ffffffff8f145060 (fs_reclaim){+.+.}, at: __fs_reclaim_acquire+0x5/0x30
[ 1334.989309]
               which lock already depends on the new lock.

[ 1334.989713]
               the existing dependency chain (in reverse order) is:
[ 1334.990370]
               -> #1 (fs_reclaim){+.+.}:
[ 1334.990684]        fs_reclaim_acquire.part.0+0x25/0x30
[ 1334.990946]        kmem_cache_alloc_node_trace+0x33/0x310
[ 1334.991215]        __get_vm_area_node+0xa0/0x310
[ 1334.991885]        vmap+0xb7/0x150
[ 1334.992370]        __bio_map_or_bounce+0xb4f/0x1230 [bcachefs]
[ 1334.992989]        bch2_bio_compress+0x313/0xe50 [bcachefs]
[ 1334.993492]        __bch2_write+0x23a5/0x37c0 [bcachefs]
[ 1334.994118]        bch2_write+0x68a/0x13b0 [bcachefs]
[ 1334.994607]        bch2_writepage_do_io.isra.0+0xea/0x190 [bcachefs]
[ 1334.995043]        bch2_writepages+0x181/0x210 [bcachefs]
[ 1334.995442]        do_writepages+0xe0/0x260
[ 1334.995842]        __writeback_single_inode+0xd0/0xd80
[ 1334.996172]        writeback_sb_inodes+0x43a/0xad0
[ 1334.996489]        __writeback_inodes_wb+0xb7/0x1e0
[ 1334.996814]        wb_writeback+0x5fd/0x8d0
[ 1334.997122]        wb_workfn+0x716/0xd70
[ 1334.997519]        process_one_work+0x7cd/0x1320
[ 1334.998551]        worker_thread+0x92/0xc30
[ 1334.998922]        kthread+0x2bc/0x3b0
[ 1334.999333]        ret_from_fork+0x3a/0x50
[ 1334.999662]
               -> #0 (&wp->lock){+.+.}:
[ 1335.000069]        __lock_acquire+0x20bc/0x49b0
[ 1335.000397]        lock_acquire+0x117/0x2f0
[ 1335.000709]        __mutex_lock+0x143/0x11b0
[ 1335.001267]        bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.001725]        __bch2_write+0x3eb/0x37c0 [bcachefs]
[ 1335.002220]        bch2_write+0x68a/0x13b0 [bcachefs]
[ 1335.002647]        bch2_writepage_do_io.isra.0+0xea/0x190 [bcachefs]
[ 1335.003031]        bch2_writepage+0x163/0x1e0 [bcachefs]
[ 1335.003399]        pageout+0x273/0x8a0
[ 1335.003849]        shrink_page_list+0x1472/0x2060
[ 1335.004258]        shrink_inactive_list+0x3b7/0xac0
[ 1335.004701]        shrink_lruvec+0x715/0xfe0
[ 1335.005087]        shrink_node+0x37d/0x16c0
[ 1335.005459]        balance_pgdat+0x5cc/0xd60
[ 1335.005883]        kswapd+0x4f6/0xaf0
[ 1335.008539]        kthread+0x2bc/0x3b0
[ 1335.010739]        ret_from_fork+0x3a/0x50
[ 1335.010968]
               other info that might help us debug this:

[ 1335.011448]  Possible unsafe locking scenario:

[ 1335.012012]        CPU0                    CPU1
[ 1335.012377]        ----                    ----
[ 1335.012663]   lock(fs_reclaim);
[ 1335.012904]                                lock(&wp->lock);
[ 1335.013267]                                lock(fs_reclaim);
[ 1335.013589]   lock(&wp->lock);
[ 1335.013805]
                *** DEADLOCK ***

[ 1335.014235] 1 lock held by kswapd0/109:
[ 1335.014457]  #0: ffffffff8f145060 (fs_reclaim){+.+.}, at: __fs_reclaim_acquire+0x5/0x30
[ 1335.014984]
               stack backtrace:
[ 1335.015428] CPU: 2 PID: 109 Comm: kswapd0 Tainted: G           OE   T 5.6.0-134515-ge1f6739c4a9f #21
[ 1335.015865] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
[ 1335.016570] Call Trace:
[ 1335.016837]  dump_stack+0x97/0xe0
[ 1335.017096]  check_noncircular+0x32e/0x3e0
[ 1335.017380]  ? unwind_next_frame+0x1be/0x18a0
[ 1335.017932]  ? print_circular_bug.isra.0+0x1e0/0x1e0
[ 1335.018224]  ? graph_lock+0x80/0x180
[ 1335.018451]  ? __lockdep_reset_lock+0x3d0/0x3d0
[ 1335.018780]  ? mark_lock+0xbc/0x1150
[ 1335.019017]  ? rcu_read_lock_bh_held+0xc0/0xc0
[ 1335.019289]  __lock_acquire+0x20bc/0x49b0
[ 1335.019525]  ? mark_held_locks+0xf0/0xf0
[ 1335.019749]  ? bch2_writepage+0x150/0x1e0 [bcachefs]
[ 1335.020141]  lock_acquire+0x117/0x2f0
[ 1335.020366]  ? bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.020626]  __mutex_lock+0x143/0x11b0
[ 1335.021002]  ? bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.021284]  ? bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.021547]  ? _raw_spin_unlock_irqrestore+0x37/0x40
[ 1335.021768]  ? mutex_trylock+0x2d0/0x2d0
[ 1335.021962]  ? lockdep_hardirqs_on+0x351/0x550
[ 1335.022179]  ? stack_depot_save+0x254/0x440
[ 1335.022463]  ? save_stack+0x4c/0x80
[ 1335.022676]  ? save_stack+0x1b/0x80
[ 1335.022870]  ? __kasan_kmalloc.constprop.0+0xc2/0xd0
[ 1335.023112]  ? kmem_cache_alloc+0xd3/0x2a0
[ 1335.023563]  ? mempool_alloc+0x102/0x2e0
[ 1335.023813]  ? bvec_alloc+0xa5/0x270
[ 1335.024170]  ? bio_alloc_bioset+0x2e5/0x4b0
[ 1335.024442]  ? __bch2_writepage+0xbc3/0x2200 [bcachefs]
[ 1335.024904]  ? bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.025211]  bch2_alloc_sectors_start+0x22b/0x1550 [bcachefs]
[ 1335.025494]  ? shrink_node+0x37d/0x16c0
[ 1335.025721]  ? mark_lock+0xbc/0x1150
[ 1335.026042]  ? mark_lock+0xbc/0x1150
[ 1335.026307]  ? bch2_writepoint_stop+0x50/0x50 [bcachefs]
[ 1335.026681]  ? mempool_alloc+0x102/0x2e0
[ 1335.026928]  ? rcu_read_lock_sched_held+0xa5/0xc0
[ 1335.027206]  __bch2_write+0x3eb/0x37c0 [bcachefs]
[ 1335.027539]  ? mempool_alloc+0x102/0x2e0
[ 1335.027752]  ? mempool_alloc+0x102/0x2e0
[ 1335.028014]  ? __lock_acquire+0x830/0x49b0
[ 1335.028286]  ? bch2_submit_wbio_replicas+0xd70/0xd70 [bcachefs]
[ 1335.028575]  ? find_held_lock+0x2d/0x110
[ 1335.028846]  ? bch2_write+0x4d1/0x13b0 [bcachefs]
[ 1335.029095]  ? lock_downgrade+0x680/0x680
[ 1335.029489]  ? lock_acquire+0x117/0x2f0
[ 1335.029823]  ? bch2_write+0x458/0x13b0 [bcachefs]
[ 1335.030118]  bch2_write+0x68a/0x13b0 [bcachefs]
[ 1335.030412]  ? __bch2_write+0x37c0/0x37c0 [bcachefs]
[ 1335.030758]  ? bch2_writepage_io_free+0x20/0x20 [bcachefs]
[ 1335.031135]  ? clear_page_dirty_for_io+0x313/0x750
[ 1335.031512]  bch2_writepage_do_io.isra.0+0xea/0x190 [bcachefs]
[ 1335.032058]  bch2_writepage+0x163/0x1e0 [bcachefs]
[ 1335.032443]  ? bch2_writepages+0x210/0x210 [bcachefs]
[ 1335.032779]  ? clear_page_dirty_for_io+0x326/0x750
[ 1335.033036]  ? find_held_lock+0x2d/0x110
[ 1335.033290]  pageout+0x273/0x8a0
[ 1335.033508]  ? shrink_slab+0x4b0/0x4b0
[ 1335.033743]  ? page_mapped+0xf3/0x270
[ 1335.033995]  ? inode_congested+0x2eb/0x580
[ 1335.034256]  shrink_page_list+0x1472/0x2060
[ 1335.034492]  ? page_evictable+0x1c0/0x1c0
[ 1335.034749]  ? mark_lock+0xbc/0x1150
[ 1335.034975]  ? do_raw_spin_lock+0x12e/0x2a0
[ 1335.035328]  ? mark_held_locks+0x67/0xf0
[ 1335.035691]  ? _raw_spin_unlock_irq+0x24/0x30
[ 1335.035960]  shrink_inactive_list+0x3b7/0xac0
[ 1335.036230]  ? move_pages_to_lru+0xcf0/0xcf0
[ 1335.036574]  ? shrink_lruvec+0xb7f/0xfe0
[ 1335.036916]  ? _raw_spin_unlock_irq+0x24/0x30
[ 1335.038847]  shrink_lruvec+0x715/0xfe0
[ 1335.039103]  ? shrink_active_list+0xbf0/0xbf0
[ 1335.039381]  ? lock_downgrade+0x680/0x680
[ 1335.039626]  ? mem_cgroup_iter+0x1ce/0xca0
[ 1335.040680]  ? mem_cgroup_iter+0x52f/0xca0
[ 1335.041019]  ? mem_cgroup_iter+0x52f/0xca0
[ 1335.041251]  ? uncharge_page+0x430/0x430
[ 1335.041607]  ? lock_acquire+0x117/0x2f0
[ 1335.042072]  ? css_next_child+0xd4/0x220
[ 1335.042454]  shrink_node+0x37d/0x16c0
[ 1335.042818]  balance_pgdat+0x5cc/0xd60
[ 1335.043150]  ? mem_cgroup_shrink_node+0x630/0x630
[ 1335.043505]  ? find_held_lock+0x2d/0x110
[ 1335.044019]  ? rcu_read_lock_sched_held+0xa5/0xc0
[ 1335.044409]  ? rcu_read_lock_bh_held+0xc0/0xc0
[ 1335.044831]  ? _raw_spin_unlock_irqrestore+0x37/0x40
[ 1335.045259]  kswapd+0x4f6/0xaf0
[ 1335.045560]  ? balance_pgdat+0xd60/0xd60
[ 1335.045926]  ? __kthread_parkme+0x51/0x100
[ 1335.046291]  ? finish_wait+0x220/0x220
[ 1335.046901]  ? lockdep_hardirqs_on+0x351/0x550
[ 1335.047380]  ? balance_pgdat+0xd60/0xd60
[ 1335.047729]  kthread+0x2bc/0x3b0
[ 1335.048038]  ? kthread_create_on_node+0xd0/0xd0
[ 1335.048428]  ret_from_fork+0x3a/0x50
[ 1337.583900] influxd invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
[ 1337.592165] CPU: 3 PID: 4367 Comm: influxd Tainted: G           OE   T 5.6.0-134515-ge1f6739c4a9f #21
[ 1337.592569] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
```

**Comment by koverstreet** [comment]
Fixed by 8b23a1b622

**Comment by marcin-github** [comment]
Issue came back. commit 2c71b506e567 
```

[44968.838421] ======================================================
[44968.838668] WARNING: possible circular locking dependency detected
[44968.838908] 5.7.0-00526-g2c71b506e567 #22 Tainted: G           OE   T
[44968.839182] ------------------------------------------------------
[44968.839425] kswapd0/110 is trying to acquire lock:
[44968.839658] ffff8e299ea5ef48 (&wp->lock){+.+.}-{3:3}, at: bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.840050]
               but task is already holding lock:
[44968.840368] ffffffffb506e580 (fs_reclaim){+.+.}-{0:0}, at: __fs_reclaim_acquire+0x5/0x30
[44968.840722]
               which lock already depends on the new lock.

[44968.841143]
               the existing dependency chain (in reverse order) is:
[44968.841481]
               -> #1 (fs_reclaim){+.+.}-{0:0}:
[44968.841783]        fs_reclaim_acquire.part.0+0x25/0x30
[44968.842003]        __alloc_pages_nodemask+0x2ad/0x3c0
[44968.842220]        __get_free_pages+0xd/0x30
[44968.842526]        __pte_alloc_kernel+0x16/0xa0
[44968.842738]        vmap_page_range_noflush+0x2a6/0x300
[44968.842963]        map_vm_area+0x31/0x40
[44968.843156]        __vmalloc_node_range+0x1e5/0x260
[44968.843372]        __vmalloc+0x30/0x40
[44968.843564]        mempool_alloc+0x62/0x170
[44968.843812]        bch2_bio_compress+0x15b/0x5a0 [bcachefs]
[44968.844061]        __bch2_write+0xdae/0x1090 [bcachefs]
[44968.844299]        bch2_writepage_do_io.isra.0+0x3b/0x60 [bcachefs]
[44968.844560]        __bch2_writepage+0x35e/0x7c0 [bcachefs]
[44968.844808]        write_cache_pages+0x18a/0x480
[44968.845042]        bch2_writepages+0x9d/0xe0 [bcachefs]
[44968.845271]        do_writepages+0x41/0xd0
[44968.845474]        __writeback_single_inode+0x57/0x570
[44968.845702]        writeback_sb_inodes+0x1fb/0x4a0
[44968.845909]        __writeback_inodes_wb+0x4c/0xc0
[44968.846124]        wb_writeback+0x2fc/0x410
[44968.846322]        wb_workfn+0x425/0x570
[44968.846522]        process_one_work+0x230/0x530
[44968.846734]        worker_thread+0x50/0x3a0
[44968.846935]        kthread+0xf6/0x130
[44968.847123]        ret_from_fork+0x3a/0x50
[44968.847340]
               -> #0 (&wp->lock){+.+.}-{3:3}:
[44968.847639]        __lock_acquire+0x121b/0x2470
[44968.847852]        lock_acquire+0xa5/0x330
[44968.848085]        __mutex_lock+0x89/0x910
[44968.848358]        bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.848676]        __bch2_write+0x144/0x1090 [bcachefs]
[44968.848858]        bch2_writepage_do_io.isra.0+0x3b/0x60 [bcachefs]
[44968.849049]        bch2_writepage+0x9c/0xc0 [bcachefs]
[44968.849209]        pageout+0xe8/0x350
[44968.849349]        shrink_page_list+0xa5b/0xc00
[44968.849547]        shrink_inactive_list+0x1e3/0x420
[44968.849747]        shrink_lruvec+0x385/0x6b0
[44968.849926]        shrink_node+0x169/0x680
[44968.850103]        balance_pgdat+0x2e2/0x620
[44968.850269]        kswapd+0x218/0x4b0
[44968.850428]        kthread+0xf6/0x130
[44968.850585]        ret_from_fork+0x3a/0x50
[44968.850744]
               other info that might help us debug this:

[44968.851051]  Possible unsafe locking scenario:

[44968.851293]        CPU0                    CPU1
[44968.851461]        ----                    ----
[44968.851628]   lock(fs_reclaim);
[44968.851779]                                lock(&wp->lock);
[44968.851962]                                lock(fs_reclaim);
[44968.852149]   lock(&wp->lock);
[44968.852299]
                *** DEADLOCK ***

[44968.852580] 1 lock held by kswapd0/110:
[44968.852748]  #0: ffffffffb506e580 (fs_reclaim){+.+.}-{0:0}, at: __fs_reclaim_acquire+0x5/0x30
[44968.857032]
               stack backtrace:
[44968.857365] CPU: 4 PID: 110 Comm: kswapd0 Tainted: G           OE   T 5.7.0-00526-g2c71b506e567 #22                                                                                              [194/1968]
[44968.857780] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
[44968.858339] Call Trace:
[44968.858575]  dump_stack+0x71/0xa0
[44968.858770]  check_noncircular+0x162/0x180
[44968.859013]  __lock_acquire+0x121b/0x2470
[44968.859235]  lock_acquire+0xa5/0x330
[44968.859455]  ? bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.859715]  ? check_noncircular+0x9b/0x180
[44968.860002]  __mutex_lock+0x89/0x910
[44968.860259]  ? bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.860509]  ? orc_find.part.0+0xf0/0x1a0
[44968.860749]  ? ret_from_fork+0x3a/0x50
[44968.860966]  ? unwind_next_frame+0x5f/0x530
[44968.861203]  ? bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.861479]  ? arch_stack_walk+0x8c/0xf0
[44968.861722]  ? bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.861993]  bch2_alloc_sectors_start+0x11e/0x860 [bcachefs]
[44968.862427]  ? alloc_list_entry+0x12/0xd0
[44968.862699]  ? atomic_dec_and_mutex_lock+0x60/0x70
[44968.863019]  __bch2_write+0x144/0x1090 [bcachefs]
[44968.863310]  ? lock_acquire+0xa5/0x330
[44968.863562]  ? down+0x12/0x50
[44968.863796]  ? find_held_lock+0x2b/0x80
[44968.864047]  ? mark_held_locks+0x2c/0x80
[44968.864273]  ? _raw_spin_unlock_irqrestore+0x37/0x40
[44968.864490]  ? lockdep_hardirqs_on+0xeb/0x190
[44968.864751]  ? bch2_write+0x304/0x530 [bcachefs]
[44968.865023]  bch2_writepage_do_io.isra.0+0x3b/0x60 [bcachefs]
[44968.865644]  bch2_writepage+0x9c/0xc0 [bcachefs]
[44968.866044]  pageout+0xe8/0x350
[44968.866347]  shrink_page_list+0xa5b/0xc00
[44968.867643]  shrink_inactive_list+0x1e3/0x420
[44968.867872]  shrink_lruvec+0x385/0x6b0
[44968.876083]  ? lock_acquire+0xa5/0x330
[44968.876297]  shrink_node+0x169/0x680
[44968.876517]  balance_pgdat+0x2e2/0x620
[44968.876765]  kswapd+0x218/0x4b0
[44968.876974]  ? finish_wait+0x80/0x80
[44968.877193]  kthread+0xf6/0x130
[44968.877417]  ? balance_pgdat+0x620/0x620
[44968.877642]  ? kthread_park+0x90/0x90
[44968.877875]  ret_from_fork+0x3a/0x50
[46164.847890] apache2 invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
[46164.848401] CPU: 3 PID: 8358 Comm: apache2 Tainted: G           OE   T 5.7.0-00526-g2c71b506e567 #22
[46164.849070] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
[46164.849591] Call Trace:
[46164.849805]  dump_stack+0x71/0xa0
[46164.850046]  dump_header+0x47/0x2e5
[46164.850283]  oom_kill_process.cold+0xb/0x10
[46164.850539]  out_of_memory+0x19a/0x530
[46164.850787]  __alloc_pages_slowpath.constprop.0+0xc49/0xf40
[46164.851083]  ? __switch_to_asm+0x34/0x70
[46164.851380]  __alloc_pages_nodemask+0x347/0x3c0
[46164.851692]  alloc_pages_vma+0xcc/0x170
[46164.851953]  __read_swap_cache_async+0x186/0x240
[46164.852233]  swapin_readahead+0x25b/0x3f0
[46164.852539]  do_swap_page+0x174/0x940
[46164.852807]  __handle_mm_fault+0x463/0xca0
[46164.853211]  handle_mm_fault+0x141/0x340
[46164.853476]  do_page_fault+0x2ae/0x674
[46164.853734]  async_page_fault+0x34/0x40
[46164.853999] RIP: 0033:0x7f40190b9fe0
[46164.854246] Code: 8d 7c 17 f8 48 8d 74 16 f8 c5 fa 7e 0f c5 fa 7e 16 c5 e9 74 d1 c5 f9 d7 c2 2d ff ff 00 00 0f 85 06 ff ff ff c3 0f 1f 44 00 00 <c5> fa 6f 16 c5 e9 74 17 c5 f9 d7 c2 2d ff ff 00 00 0f 85
e9 fe ff
[46164.855043] RSP: 002b:00007ffc2d42c8e8 EFLAGS: 00010246
[46164.855342] RAX: 0000560775718460 RBX: 0000560775718770 RCX: 000000000000d99a
[46164.855756] RDX: 0000000000000010 RSI: 00007f40191d6180 RDI: 00005607757184a8
[46164.856162] RBP: 00007ffc2d42ca00 R08: 00000000000000a0 R09: 0000560775bcab20
[46164.856591] R10: 0000000000080000 R11: 0000000000000246 R12: 0000000000000000
[46164.857024] R13: 0000560775bcab20 R14: 0000560775bcab20 R15: 000000000000000a
[46164.861460] Mem-Info:
[46164.861932] active_anon:1016841 inactive_anon:169478 isolated_anon:224
                active_file:506 inactive_file:429 isolated_file:123
                unevictable:9386 dirty:6 writeback:0 unstable:0
                slab_reclaimable:13076 slab_unreclaimable:30673
                mapped:5072 shmem:11260 pagetables:5669 bounce:0
                free:20433 free_pcp:1879 free_cma:0
[46164.863825] Node 0 active_anon:4067168kB inactive_anon:678304kB active_file:1292kB inactive_file:924kB unevictable:37544kB isolated(anon):640kB isolated(file):492kB mapped:19112kB dirty:24kB writeback:0$
B shmem:45040kB writeback_tmp:0kB unstable:0kB all_unreclaimable? no
[46164.864935] Node 0 DMA free:15908kB min:20kB low:32kB high:44kB reserved_highatomic:0KB active_anon:0kB inactive_anon:0kB active_file:0kB inactive_file:0kB unevictable:0kB writepending:0kB present:15992$
B managed:15908kB mlocked:0kB kernel_stack:0kB pagetables:0kB bounce:0kB free_pcp:0kB local_pcp:0kB free_cma:0kB
[46164.866707] lowmem_reserve[]: 0 847 9551 9551
[46164.867083] Node 0 DMA32 free:36768kB min:1104kB low:1972kB high:2840kB reserved_highatomic:0KB active_anon:339944kB inactive_anon:36kB active_file:36kB inactive_file:36kB unevictable:0kB writepending:0$
B present:1032012kB managed:606356kB mlocked:0kB kernel_stack:672kB pagetables:308kB bounce:0kB free_pcp:672kB local_pcp:16kB free_cma:0kB
[46164.868569] lowmem_reserve[]: 0 0 8704 8704
[46164.868841] Node 0 Normal free:31492kB min:11368kB low:20280kB high:29192kB reserved_highatomic:2048KB active_anon:3734580kB inactive_anon:670884kB active_file:2152kB inactive_file:2084kB unevictable:37$
44kB writepending:0kB present:8912896kB managed:6691288kB mlocked:37544kB kernel_stack:5600kB pagetables:22368kB bounce:0kB free_pcp:5320kB local_pcp:1456kB free_cma:0kB
[46164.870264] lowmem_reserve[]: 0 0 0 0
[46164.870622] Node 0 DMA: 1*4kB (U) 0*8kB 0*16kB 1*32kB (U) 2*64kB (U) 1*128kB (U) 1*256kB (U) 0*512kB 1*1024kB (U) 1*2048kB (M) 3*4096kB (M) = 15908kB
[46164.871217] Node 0 DMA32: 132*4kB (UM) 190*8kB (UE) 176*16kB (UME) 155*32kB (UME) 99*64kB (UME) 63*128kB (UME) 28*256kB (UME) 9*512kB (UM) 0*1024kB 0*2048kB 0*4096kB = 36000kB
[46164.871930] Node 0 Normal: 834*4kB (MH) 820*8kB (UMEH) 605*16kB (UMEH) 240*32kB (UME) 26*64kB (UMH) 4*128kB (MH) 2*256kB (H) 2*512kB (H) 0*1024kB 0*2048kB 0*4096kB = 30968kB
[46164.872711] Node 0 hugepages_total=0 hugepages_free=0 hugepages_surp=0 hugepages_size=1048576kB
[46164.873281] Node 0 hugepages_total=200 hugepages_free=142 hugepages_surp=0 hugepages_size=2048kB
[46164.873790] 14463 total pagecache pages
[46164.874042] 143 pages in swap cache
[46164.874463] Swap cache stats: add 271996, delete 271781, find 77671/79742
[46164.874865] Free swap  = 4kB
[46164.875090] Total swap = 1060856kB
[46164.875386] 2490225 pages RAM
[46164.875612] 0 pages HighMem/MovableOnly
[46164.875883] 661837 pages reserved
[46164.876145] 1024 pages cma reserved
[46164.876422] Tasks state (memory values in pages):
[46164.876798] [  pid  ]   uid  tgid total_vm      rss pgtables_bytes swapents oom_score_adj name
[46164.877300] [   1489]     0  1489     3236      528    45056      153             0 systemd-udevd
[...]
```

**Comment by koverstreet** [comment]
There were more allocations - should be fixed now, by 152274b005
